### PR TITLE
Improve integration tests, small refactor

### PR DESCRIPTION
**Pull request by eskombro**
This PR provides a wider set of integration tests (Client, Index, Documents and Updates tests).

To run the tests in your local machine you can simply run

`gradle cleanTest integrationTest`

It also isolates Search and Update classes (and it's corresponding methods) from the Documents class.

It doesn't cover search tests as, IMO, the search needs to be refactored (search will be addressed in a different issue/PR), but it isn't extremely urgent as this SDK will mostly be used for back-end purposes!

### CODE REVIEW

File: src/test/java/com/meilisearch/integration/DocumentsTest.java [hunk 1]
```diff
@@ -78,5 +77,143 @@ public void testAddDocumentsMultiple() throws Exception {
...
+		updateInfo = this.gson.fromJson(
+			index.deleteDocuments(),
+			UpdateStatus.class
+		);
+		index.waitForPendingUpdate(updateInfo.getUpdateId());
+
+		movies = this.gson.fromJson(index.getDocuments(), Movie[].class);
+		assertEquals(0, movies.length);
+	}
+
```
Comment (curquiza) [inline src/test/java/com/meilisearch/integration/DocumentsTest.java:217]: There is a missing test for documents deletion. I don't know if `deleteDocuments` is the [delete-batch route](https://docs.meilisearch.com/references/documents.html#delete-documents) or the [delete-all one](https://docs.meilisearch.com/references/documents.html#delete-documents) but there is one missing in the tests.

File: src/test/resources/movies.json [hunk 2]
```diff
@@ -1,135 +1 @@
...
...": "2017-11-15", "language": "en", "genres": ["Action", "Adventure", "Fantasy", "Science Fiction"]}, {"id": 671, "title": "Harry Potter and the Philosopher's Stone", "poster": "https://image.tmdb.org/t/p/original/wuMc08IPKEatf9rnMNXvIDxqP4W.jpg", "overview": "Harry Potter has lived under the stairs at his aunt and uncle's house his whole life. But on his 11th birthday, he learns he's a powerful wizard -- with a place waiting for him at the Hogwarts School of Witchcraft and Wizardry. As he learns to harness his newfound powers with the help of the school's kindly headmaster, Harry uncovers the truth about his parents' deaths -- and about the villain who's to blame.", "release_date": "2001-11-16", "language": "en", "genres": ["Adventure", "Fantasy", "Family"]}, {"id": 509585, "title": "7500", "poster": "https://image.tmdb.org/t/p/original/hL2uecLh2rTTbuVbOriXP0PhqIJ.jpg", "overview": "When terrorists try to seize control of a Berlin-Paris flight, a soft-spoken young American co-pilot struggles to save the lives of the passengers and crew while forging a surprising connection with one of the hijackers.", "release_date": "2019-12-26", "language": "en", "genres": ["Thriller", "Drama"]}]
```
Comment (curquiza) [inline src/test/resources/movies.json:1]: Why doing a single line? It was readable before and more maintainable if we have to improve the tests so the dataset in the future

**Review by curquiza — COMMENTED** [review]
Useful tests addition!
Some comments 🙂

File: src/main/java/com/meilisearch/sdk/Search.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	String search(String uid,
+				  String q,
+				  int offset,
+				  int limit,
+				  String attributesToRetrieve,
+				  String attributesToCrop,
+				  int cropLength,
+				  String attributesToHighlight,
+				  String filters,
+				  boolean matches
```
[the remark this answers is not in the document]
Comment (eskombro) [reply src/main/java/com/meilisearch/sdk/Search.java:25]: It must be implemented, but I think the whole search needs to be discussed/implemented, so it won't be part of this PR

File: src/test/java/com/meilisearch/integration/DocumentsTest.java (hunk 1 above)
Comment (eskombro) [reply src/test/java/com/meilisearch/integration/DocumentsTest.java:217]: Both links point to the same place 😄 
But I created an issue https://github.com/meilisearch/meilisearch-java/issues/42

File: src/test/resources/movies.json (hunk 2 above)
Comment (eskombro) [reply src/test/resources/movies.json]: Just made it readable again ;)

Comment (curquiza) [reply src/test/resources/movies.json]: Awesome!

File: src/test/java/com/meilisearch/integration/DocumentsTest.java (hunk 1 above)
Comment (curquiza) [reply src/test/java/com/meilisearch/integration/DocumentsTest.java:217]: Haha my bad! Your comment and the issue answer perfectly my question!

Comment (eskombro) [reply src/test/java/com/meilisearch/integration/DocumentsTest.java:217]: No, it was you who inspired the issue creation 🎉

**Review by niemannd — DISMISSED** [review]
looks good from my pov

File: src/test/resources/movies.json [hunk 4]
```diff
@@ -1,135 +1,400 @@
...
+	   "overview":"When terrorists try to seize control of a Berlin-Paris flight, a soft-spoken young American co-pilot struggles to save the lives of the passengers and crew while forging a surprising connection with one of the hijackers.",
+	   "release_date":"2019-12-26",
+	   "language":"en",
+	   "genres":[
+		  "Thriller",
+		  "Drama"
+	   ]
 	}
-]
+ ]
```
Comment (curquiza) [inline src/test/resources/movies.json:400]:
```suggestion
]
```

**Review by curquiza — APPROVED** [review]
The worst suggestion of change ever!
Otherwise, I approve!!! 🚀

**Review by curquiza — APPROVED** [review]

**Pull request merged**